docs: Solution design for the "Hausanschluss" Bundle (#241)
eclipse-tractusx/tractus-x-umbrella: docs/concept/TAP7.1-solution-design.md

    DataConsumer --> EDC
    EnablementProvider --> DataConsumer
    EnablementProvider --> DataProvider

````

#### Deployment strategies

##### Role: Data Provider

The standard data provider role consists of a EDC, DTR and Submodel Server Bundles. EDC uses the Identity Wallet Bundle.

````mermaid
flowchart BT
    A["Capability Bundle: Identity & Trust"]
    B["Capability Bundle: Data Persistence Layer"]
    C["Capability Bundle: Dataspace Connector"]
    D["Capability Bundle: Digital Twin"]
    E["Application to be tested/developed"]
    E --> B
    E --> C
    E --> D
    C --> A
````

##### Role: Data Consumer

The standard data consumer role consists of only the EDC Bundle and the Identity Wallet Bundle.

````mermaid
flowchart BT
    A["Capability Bundle: Identity & Trust"]
    B["Capability Bundle: Dataspace Connector"]
    C["Application to be tested/developed"]
    C --> B
    B --> A

````

##### Role: Enablement Provider

The Enablement Provider roles consists of both the Data Consumer and Data Provider bundles. This allows provisioning of
data and the consumtion of the provisioned data by a application.

````mermaid
flowchart BT

>     A["Capability Bundle: Identity & Trust"]

Author: added the actors

> # [TAP 7.1] Solution Design Hausanschluss Bundle Konzept

Reviewer: @ds-jhartmann 
unique naming EDC "Hausanschluss" Bundle or Hausanschluss Bundle (in the whole doument)

# 1 Business Case

## 1.1 Executive Summary

The EDC "Hausanschluss" Bundle aims to simplify the installation of the enablement services Eclipse Dataspace
Connector (EDC), Digital Twin Registry (DTR), and their supporting components. By developing a flexible, parameterizable
installation package, the project seeks to reduce deployment complexity and lower barriers to participation in the
Catena‑X data space. This bundle will enable independent component updates and support a modular "bring your own"
approach, ultimately fostering a more efficient and accessible ecosystem for diverse participants—such as testers,
external developers, and small and medium-sized enterprises.

## 1.2 Business Need

Current deployment methods for data space integration are complex and require significant manual effort. This PoC
addresses the need to:

- Reduce Complexity: Simplify the deployment topology by bundling components into capabilities.
- Automate Deployment: Provide an automated installation package using Kubernetes and HELM.

> - Reduce Configuration Efforts: Streamline and reduce the number of configuration steps required for integration.

Author: rephrased to a more qualitative description

- Extend Documentation: Provide a detailed documentation tailored to the target groups.
- Enable Extension: Design the system to allow the seamless addition of new components or capabilities.

## 1.3 Objectives

> - Simplify Deployment and Maintenance: Develop an installation package that minimizes manual configuration through

Reviewer: Are all these objectives covered in the concept?

  automation.
- Ensure Modularity: Structure the bundle so that components can be independently installed, updated, or replaced
  without
  affecting other services.

> - Facilitate Flexibility: Implement a "bring your own" model for key components, such as EDC, PostgreSQL, and HashiCorp

Author: thats in the next line

  Vault, to allow tailored deployments that meet specific needs.
- Increase Accessibility: Lower entry barriers for new participants by reducing technical complexity and operational
  dependencies.
- Scalability: Structure the bundle so that it can be deployed multiple times.

## 1.4 Expected Deliverables

> - PoC Documentation: A concept and architecture overview highlighting the modular design.

Reviewer: I think user or developer documentation would be more clear.

Author: User documentation is the last point in this list already

> - PoC Documentation: A concept and architecture overview highlighting the modular design.

Reviewer: -> Is this document providing this service? "architecture overview highlighting the modular design"

- Modular Installation Package: A set of HELM charts for deploying the enablement service bundles:

>     - EDC Bundle: Including the Eclipse Dataspace Connector, EDC PostgreSQL, and HashiCorp Vault.

Reviewer: Are we willing to create a HELM for the HashiCorp Vault

Author: no, there is no need for that. The table was not migrated correctly which lead to some missleading content. This is now fixed

## 1.6. Dependencies

- Interoperability with the CATENA‑X standards.

## 1.7. Success Criteria

- Technology Readiness Level: Achievement of a Technology Readiness Level (TRL) of 3, as specified in the requirements.
- Functional PoC Deployment: Demonstrated automated, modular installation with reduced manual steps.
- Simplified Configuration: Clear evidence of reduced configuration complexity.
- Positive Feedback: End-user validation from a select group of demo participants and the Tractus-X community.
- Documentation Completeness: Availability of concise deployment and user guides.

## 1.8. Out of Scope

- Full Production Readiness: This PoC is not intended to be a production-ready solution.
- Core Services Exclusion: Core services are not part of the Hausanschluss Bundles.
- App Provider Role: The role of App Provider is considered out of scope for this deployment.
- Alternative Deployment Techniques: While the evaluation of alternative deployment methods is acknowledged, the primary
  focus remains on Kubernetes and HELM-based deployment.

# 2 Requirements Analysis

## 2.1 Assumptions

## 2.2 Requirements Catalog

> | ID             | Requirement                                                                                                                                                                                                                                                                                                                                               | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | Deliverables / Acceptance Criteria                                                                                                           |

Reviewer: Is this confirmed that ONLY MUST criterias are listed here?

Author: yes, these are all the must criterias

# 3 AS-IS Analysis

## 3.1 People

### 3.1.1 Target Group Analysis

See "4.1.1 To-be Target Group Analysis" below. No deviations between As-is and To-be target groups have been identified.

## 3.2 Process

### 3.2.1 Use Case View (high-level)

Umbrella Chart is intended for the following Use Cases:

| Actor             | Use case                                                                                             |
|-------------------|------------------------------------------------------------------------------------------------------|
| Product Developer | Sandbox Environments: Create local environments to evaluate and experiment with Catena-X components. |
| Product Developer | Development: Provide a unified setup for contributors and developers working on Tractus-X projects.  |

> | Product Tester    | Testing: Run end-to-end tests for Catena-X services.                                                 |

Reviewer: Product Manager is one more important Target Group we should add. Evaluating the data space .... This could be added later....

Author: added the PM to use cases

            DiscoveryFinder["Discovery Finder"]
            SDFactory["SD-Factory"]
            SSICredentialIssuer["SSI Credential Issuer"]
            SemanticHub["Semantic Hub"]
            BPDM["BPDM"]
            DataConsumer["Data Consumer"]
            DataProvider["Data Provider"]
            BDRS["BDRS"]
            SSIWalletStub["SSI Wallet Stub"]
        end
        subgraph "tx-data-provider"
            EDC["EDC"]
            DTR["DTR"]
            SubmodelServer["Submodel Server"]
        end
    end

    ChartYaml --> Portal
    ChartYaml --> CentralIdp
    ChartYaml --> DiscoveryFinder
    ChartYaml --> SDFactory
    ChartYaml --> SSICredentialIssuer
    ChartYaml --> SemanticHub
    ChartYaml --> BPDM
    ChartYaml --> DataConsumer
    ChartYaml --> DataProvider
    ChartYaml --> BDRS
    ChartYaml --> SSIWalletStub
    DataConsumer --> EDC
    DataProvider --> EDC
    DataProvider --> DTR
    DataProvider --> SubmodelServer
````

#### Existing Umbrella Chart Structure

# 4 TO-BE Design

## 4.1 People

### 4.1.1 Target Group Analysis

Overall data space roles: https://catena-x.net/en/1/catena-x-introduce-implement/how-to-operate-catena-x

> The relevant data space roles for us under requirement 7.1.1.1 are:

Reviewer: ```suggestion
The relevant data space roles for us under requirement 7.1.1.1 are:
```
```suggestion
The relevant data space roles for TAP 7.1 are:
```

- Data Consumer
- Data Provider
- Enablement Service Provider

Scope: Umbrella HELM Charts

> - Enablement Services: These services are included under the umbrella of HELM charts, focusing on enabling

Reviewer: This is still a mixup between in scope and out of scope. I have already addressed this with a comment.

> | 1   | Developers<ul><li>External Developers (not Tractus-X developers): Utilize the chart for demonstration or evaluation purposes.</li><li>Experienced software and product developers who are developing a business app for a CATENA-X data space. They need to develop the app against the interfaces of the data space and integrate them accordingly.</li><li>They emphasize efficient and cost-effective implementation:<ul><li>Cost Efficiency: They do not want to use expensive SaaS solutions.</li><li>Technical Independence: They lack the knowledge and capacity to deploy the enablement and core services of the data space themselves.</li><li>Focus on Development: They concentrate on application development without delving deeply into the deployment and operation of the underlying infrastructure.</li></ul></li><li>Out of Scope: Developers running tests in test runners are not considered within this scope.</li></ul> | Data Consumer <br/> Data Provider <br/> Enablement Service Provider |                                                                                                                                                                                                                                                                                                         | <ul><li>Needs / sources of information<ul><li><a href="https://eclipse-tractusx.github.io/Kits/">Kit documentation tractus-x</a> for interaction purposes</li><li><a href="https://catenax-ev.github.io/">Catena-X doucmentation</a> for standards and regulations</li><li><a href="https://github.com/eclipse-tractusx/tractus-x-umbrella">Umbrella documentation</a> for umbrella component information</li></ul></li><li>Challenges → Solution<ul><li>Decentralization e.g. Where are other systems? How to interact with them? → Solution: Umbrella charts and bundles for deployment</li><li>Complex setup → Solution: delivery of small capability bundles</li></ul></li></ul> |

Reviewer: Please avoid HTML Tags as these are not TRG conform

Author: Unfortunately, that is the only possibility to include a list in the table.

## 4.2 Process

### 4.2.1 Use Case View (high-level)

> | Use Case                                | Description                                                                                                       | Actors            |

Reviewer: ## Gruppe 1: Produktentwickler

| ID  | Name                                      | Akteur            | Beschreibung |
|-----|-------------------------------------------|-------------------|--------------|
| UC1 | Configure HELM Chart                      | Product Developer | The product developer configures the HELM Chart for the specific environment. |
| UC2 | Configure Secrets                         | Product Developer | The product developer manages and configures secrets. |
| UC3 | Execute Deployment Pipeline               | Product Developer | The product developer successfully executes the deployment pipeline for the HELM Chart. |
| UC4 | Integrate EDC, DTR, and Identity Wallet   | Product Developer | The actors integrate the components EDC, Digital Twin Registry, and Identity Wallet for seamless interoperability. |
| UC5 | Set Up Connection Infrastructure         | Product Developer | The actor sets up the connection infrastructure for linking to the central data space infrastructure. |
| UC6 | Maintain Infrastructure Module            | Product Developer | The product developer maintains the modular architecture of the HELM Chart. |
| UC7 | Ensure Scalability                        | Product Developer | The product developer ensures scalability to support growing numbers of users and data volumes. |
| UC8 | Sandbox Environments                      | Product Developer | The product developer creates local environments to evaluate and experiment with Catena-X components. |

## Gruppe 2: Produkttester

| ID  | Name          | Actor         | Description|
|-----|--------------|---------------|--------------|
| UC9 | E2E Testing | Product Tester | The product tester runs end-to-end tests for Catena-X services. |

Author: added

| Integrate EDC, DTR, and Identity Wallet | The actors integrate the components EDC, Digital Twin Registry, and Identity Wallet for seamless interoperability | Product Developer |
| Set Up Connection Infrastructure        | The actor sets up the connection infrastructure for linking to the central data space infrastructure.             | Product Developer |
| Maintain Infrastructure Module          | The product developer maintains the modular architecture of the HELM Chart.                                       | Product Developer |
| Ensure Scalability                      | The product developer ensures scalability to support growing numbers of users and data volumes.                   | Product Developer |
| Sandbox Environments                    | The product developer creates local environments to evaluate and experiment with Catena-X components.             | Product Developer |
| E2E Testing                             | The product tester runs end-to-end tests for Catena-X services.                                                   | Product Tester    |

> 

Reviewer: %%{init: {'theme': 'base', 'themeVariables': {'actorBorder':'#C6C6C6', 'actorBkg':'#C6C6C6', 'actorTextColor': '#000', 'primaryTextColor': '#000', 'edgeLabelBackground':'#f5f5f5', 'tertiaryColor': '#D7E9F4', 'tertiaryBorderColor': '#181818', 'tertiaryTextColor': '#000'}}}%%
usecaseDiagram
    actor "Product Developer" as PD
    actor "Product Tester" as PT

    rectangle UC {
        usecase UC1 as "Configure HELM Chart"
        usecase UC2 as "Configure Secrets"
        usecase UC3 as "Execute Deployment Pipeline"
        usecase UC4 as "Integrate EDC, DTR, and Identity Wallet"
        usecase UC5 as "Set Up Connection Infrastructure"
        usecase UC6 as "Maintain Infrastructure Module"
        usecase UC7 as "Ensure Scalability"
        usecase UC8 as "Sandbox Environments"
        usecase UC9 as "E2E Testing"
    }

    PD --> UC1
    PD --> UC2
    PD --> UC3
    PD --> UC4
    PD --> UC5
    PD --> UC6
    PD --> UC7
    PD --> UC8
    PT --> UC9

Author: Diagram does not render like this. I added a slightly adjusted version

### 4.2.2 Business Process (SCC Artifact)

#### Chart Release Process

Charts will be released using the already existing workflow in the Umbrella Chart repository. The workflow will

> automatically release a Chart, once it is pushed to the "charts/" directory with a new Chart version.

Reviewer: Any more detail info how this is versioned or why.

Author: added a remark to semantic versioning

data and the consumtion of the provisioned data by a application.

````mermaid
flowchart BT
    A["Capability Bundle: Identity & Trust"]
    B["Capability Bundle: Data Persitance Layer"]
    C["Application to be tested/developed"]
    D["Capability Bundle: Digital Twin"]
    E["Capability Bundle: Dataspace Connector (Provider)"]
    F["Capability Bundle: Dataspace Connector (Consumer)"]
    C --> B
    C --> D
    C --> E
    C --> F
    E --> A
    F --> A
````

##### Bring you own

Each Bundle can be deactivated to use your own component. This requires configuration of the component specific
parameters in the other components.

In this example this would be the configuration of the Identity Wallet within the EDC.

````mermaid
flowchart BT
    A["Application to be tested/developed"]
    B["Capability Bundle: Dataspace Connector"]
    C["<s>Capability Bundle: Identity & Trust</s>"]
    D["Own Identity Wallet"]
    A --> B
    B -- identitywallet - bundle . enabled = false --> C
    B -- configure --> D

````

## 4.4 System

### 4.4.1 Context Delimitation / 4.4.2 Interface Documentation

> 

Reviewer: Any kind of prosa documentation to the chart? Explaining the chart and what we will change here?

Author: added

````mermaid
flowchart LR
    Dataprovider["Dataprovider"] -->|stores data| SubmodelServer["Submodel Server"]
    Dataprovider -->|creates twins| DTR["DTR"]
    Dataprovider -->|creates contracts| EDC["EDC"]
    Dataconsumer["Dataconsumer"] -->|negotiate contracts| EDC
    DTR -->|needs| IdentityWallet["Identity Wallet"]
    IdentityWallet -->|needs| Postgres["PostgreSQL Database"]
````

# 5 Build

## 5.1 DevOps Setup Definition

> ### 5.1.1 Tool Chain Definition

Reviewer: Are we sure to deliver all of this? Is this mentioned in the requirements or do we extend the scope?

Author: Those are the tools we need to build and publish helm charts. Most of it is already there

##### Version Control & Source Code Management

- **Git & GitHub**: Management of the source code, the Helm Charts, and associated documentation in a centralized
  repository.

##### Build and CI/CD Tools

- **GitHub Actions**: Automated workflows for linting, testing, packaging, and releasing the Helm Charts.
- **Helm CLI**: For validating, packaging, and deploying the charts.
- **Helm Lint**: Performing static analyses to ensure compliance with chart standards.

##### Testing and Validation Tools

> - **chart-testing Tool**: Validation of the Helm Charts based on Kubernetes schemas.

Reviewer: ```suggestion
- **Chart-testing Tool**: Validation of the Helm Charts based on Kubernetes schemas.
```

> - **chart-testing Tool**: Validation of the Helm Charts based on Kubernetes schemas.

Reviewer: Is this KICS and Trivy or do we extend the scope?

Author: This is not KICS nor Trivy. This tool is just for testing helm charts

## 1.3 Objectives

- **Simplify Deployment and Maintenance:** Develop an installation package that minimizes manual configuration through
  automation.
- **Ensure Modularity:** Structure the bundle so that components can be independently installed, updated, or replaced
  without affecting other services.
- **Facilitate Flexibility:** Implement a "bring your own" model for key components, such as EDC, PostgreSQL, and HashiCorp
  Vault, to allow tailored deployments that meet specific needs.
- **Increase Accessibility**: Lower entry barriers for new participants by reducing technical complexity and operational
  dependencies.
- **Scalability:** Structure the bundles so that they can be deployed multiple times.

## 1.4 Expected Deliverables

- **PoC Documentation:** A concept and architecture overview highlighting the modular design.
- **Modular Installation Package:** A set of HELM charts for deploying the enablement service bundles:
    - **Dataspace Connector:** Including the Eclipse Dataspace Connector, EDC PostgreSQL, and HashiCorp Vault.
    - **Digital Twin Bundle:** Featuring components such as the Decentral DT Registry (with “bring your own” PostgreSQL)

>     - **Data Persitence Layer Bundle:** Including a simple Submodel Server to store semantic data.

Reviewer 2: ```suggestion
    - **Data Persistence Layer Bundle:** Including a simple Submodel Server to store semantic data.
```

> | 7.1.1.4        | Modularity & Maintainability: No dependencies between Capability Bundles                                                                                                                                                                                                                                                                             | Unique assignment of components and services to a defined Capability Bundle. <br>Each component and service should be assigned to only one Capability Bundle. <br>Dependencies between Capability Bundles are allowed.                                                                                                                                                                                                                                                                                           | Components and Services should only be part of one bundle. <br>Each bundle must be deployable in isolation and can be installed, updated, or removed independently. <br>Capability Bundles must be organized in separate HELM sub‑charts. <br>Circular dependencies between bundles must be avoided.                                                                                                                 |

Author: Thanks for the hint. You're right, this description contradicts itself. I added a updated version

> | 7.1.5.1        | Re-activate Umbrella HELM Charts to the latest release state                                                                                                                                                                                                                                                                                         | The Enablement Services components of the Umbrella HELM Charts must be updated to version R24.12. This includes the following upgrades: <br> Upgrade DTR to version R24.12 (#188) <br>Upgrade EDCs to version R24.12 (#193) <br>Update SSI-DIM-Wallet-Stub to version 0.1.8 (#206)                                                                                                                                                                                                                               | The HELM Chart will then contain the updated versions of the components. <br>After the upgrade, the components (DTR, EDCs, SSI‑DIM‑Wallet‑Stub) run stably without functional impairments. <br>The deployment pipeline for the HELM Chart executes successfully <br>Relevant tests & smoke tests (e.g., integration tests) confirm the migration.                                                                    |

Reviewer 3: ```suggestion
| 7.1.5.1        | Re-activate Umbrella HELM Charts to the latest release state                                                                                                                                                                                                                                                                                         | The Enablement Services components of the Umbrella HELM Charts must be updated to version R24.12. This includes the following upgrades: <br> Upgrade DTR to version R24.12 (#188) <br>Upgrade EDCs to version R24.12 (#193) <br>Update SSI-DIM-Wallet-Stub to version 0.1.8 (#206)                                                                                                                                                                                                                               | The HELM Chart will then contain the updated versions of the components. <br>After the upgrade, the components (DTR, EDCs, SSI‑DIM‑Wallet‑Stub) run stably without functional impairments. <br>The deployment pipeline for the HELM Chart executes successfully. <br>Relevant tests & smoke tests (e.g., integration tests) confirm the migration.                                                                    |
```

> | 7.1.5.2        | Hierarchical structuring and bundling of Enablement Service components in the Umbrella HELM Chart                                                                                                                                                                                                                                                    | The Umbrella HELM Chart should be structured hierarchically so that related Enablement Service components are grouped into bundles. <br>The Umbrella HELM Chart must have a clear hierarchical structure <br> Each Enablement Service component must be assigned to a unique bundle.<br>No circular dependencies between bundles are allowed.                                                                                                                                                                    | The chart is modular and divided into defined bundles. <br>Each bundle can be deployed and maintained independently.                                                                                                                                                                                                                                                                                                 |

Reviewer 3: ```suggestion
| 7.1.5.2        | Hierarchical structuring and bundling of Enablement Service components in the Umbrella HELM Chart                                                                                                                                                                                                                                                    | The Umbrella HELM Chart should be structured hierarchically so that related Enablement Service components are grouped into bundles. <br>The Umbrella HELM Chart must have a clear hierarchical structure. <br> Each Enablement Service component must be assigned to a unique bundle.<br>No circular dependencies between bundles are allowed.                                                                                                                                                                    | The chart is modular and divided into defined bundles. <br>Each bundle can be deployed and maintained independently.                                                                                                                                                                                                                                                                                                 |
```

> | 7.1.6.1        | POC                                                                                                                                                                                                                                                                                                                                                  | Creation of a POC for the "Hausanschluss" Bundle                                                                                                                                                                                                                                                                                                                                                                                                                                                                 | POC implementation is completed. <br>POC has been presented in a demo                                                                                                                                                                                                                                                                                                                                                |

Reviewer 3: ```suggestion
| 7.1.6.1        | POC                                                                                                                                                                                                                                                                                                                                                  | Creation of a POC for the "Hausanschluss" Bundle                                                                                                                                                                                                                                                                                                                                                                                                                                                                 | POC implementation is completed. <br>POC has been presented in a demo.                                                                                                                                                                                                                                                                                                                                                |
```

## 3.2 Process

### 3.2.1 Use Case View (high-level)

Umbrella Chart is intended for the following Use Cases:

| Actor             | Use case                                                                                             |
|-------------------|------------------------------------------------------------------------------------------------------|
| Product Developer | Sandbox Environments: Create local environments to evaluate and experiment with Catena-X components. |
| Product Developer | Development: Provide a unified setup for contributors and developers working on Tractus-X projects.  |
| Product Tester    | Testing: Run end-to-end tests for Catena-X services.                                                 |
| Product Manager   | Evaluate the dataspace to assess if their product can be used within Catena-X.                       |

## 3.4 System

### 3.4.1 Context Delimitation

#### Existing Umbrella Chart Structure

````mermaid
flowchart
    subgraph "Umbrella Chart"
        ChartYaml["Chart.yaml"]
        subgraph charts
            Portal["Portal"]

>             CentralIdp["CentralIdp"]

Reviewer 3: ```suggestion
            CentralIdp["CentralIdp"]
            SharedIdp["SharedIdp"]
```

            DiscoveryFinder["Discovery Finder"]
            SDFactory["SD-Factory"]
            SSICredentialIssuer["SSI Credential Issuer"]
            SemanticHub["Semantic Hub"]
            BPDM["BPDM"]
            DataConsumer["Data Consumer"]
            DataProvider["Data Provider"]
            BDRS["BDRS"]
            SSIWalletStub["SSI Wallet Stub"]
        end
        subgraph "tx-data-provider"
            EDC["EDC"]
            DTR["DTR"]
            SubmodelServer["Submodel Server"]
        end
    end

    ChartYaml --> Portal

>     ChartYaml --> CentralIdp

Reviewer 3: ```suggestion
    ChartYaml --> CentralIdp
    ChartYaml --> SharedIdp
```

    ChartYaml --> DiscoveryFinder
    ChartYaml --> SDFactory
    ChartYaml --> SSICredentialIssuer
    ChartYaml --> SemanticHub
    ChartYaml --> BPDM
    ChartYaml --> DataConsumer
    ChartYaml --> DataProvider
    ChartYaml --> BDRS
    ChartYaml --> SSIWalletStub
    DataConsumer --> EDC
    DataProvider --> EDC
    DataProvider --> DTR
    DataProvider --> SubmodelServer
````

# 4 TO-BE Design

## 4.1 People

### 4.1.1 Target Group Analysis

> Overall data space roles: https://catena-x.net/en/1/catena-x-introduce-implement/how-to-operate-catena-x

Reviewer 3: Currently, the URL directs to a 404 Not Found page.
![image](https://github.com/user-attachments/assets/aa84a6a2-7e7d-4fcb-a1cc-d02705034ac8)

Author: fixed

> Overall data space roles: https://catena-x.net/en/1/catena-x-introduce-implement/how-to-operate-catena-x

Author: fixed

        subgraph charts
            Portal["Portal"]
            CentralIdp["CentralIdp"]
            SharedIdp["SharedIdp"]
            DiscoveryFinder["Discovery Finder"]
            SDFactory["SD-Factory"]
            SSICredentialIssuer["SSI Credential Issuer"]
            SemanticHub["Semantic Hub"]
            BPDM["BPDM"]
            DataConsumer["Data Consumer"]
            DataProvider["Data Provider"]
            BDRS["BDRS"]
            SSIWalletStub["SSI Wallet Stub"]
        end
        subgraph "tx-data-provider"
            EDC["EDC"]
            DTR["DTR"]
            SubmodelServer["Submodel Server"]
        end
    end

    ChartYaml --> Portal
    ChartYaml --> CentralIdp
    ChartYaml --> SharedIdp
    ChartYaml --> DiscoveryFinder
    ChartYaml --> SDFactory
    ChartYaml --> SSICredentialIssuer
    ChartYaml --> SemanticHub
    ChartYaml --> BPDM
    ChartYaml --> DataConsumer
    ChartYaml --> DataProvider
    ChartYaml --> BDRS
    ChartYaml --> SSIWalletStub
    DataConsumer --> EDC
    DataProvider --> EDC
    DataProvider --> DTR
    DataProvider --> SubmodelServer
````

# 4 TO-BE Design

## 4.1 People

### 4.1.1 Target Group Analysis

> Overall data space roles: https://catena-x.net/en/1/catena-x-introduce-implement/how-to-operate-catena-x

Author: ```suggestion
Overall data space roles: https://catenax-ev.github.io/docs/operating-model/who-roles-in-the-catena-x-ecosystem
```
